src/nix/devcontainer-feature.json

```diff
@@ -43,5 +43,12 @@
     "containerEnv": {
         "PATH": "/nix/var/nix/profiles/default/bin:/nix/var/nix/profiles/default/sbin:${PATH}"
     },
+    "mounts": [
```

Comment:
Looks like this is causing multiple test failures, https://github.com/devcontainers/features/actions/runs/10992584155/job/30536518797?pr=1127 

Can you take a look? thanks!

Reply:
I think I mounted too deep, `/nix/store` instead of `/nix`. Let's try that again.

Reply:
Not helping 😞 see https://github.com/devcontainers/features/actions/runs/11014623966/job/30596929100?pr=1127

Reply:
Alright - sorry for keeping you busy. I'll dig deeper locally.

Reply:
New approach: Persistent nix-stores per devcontainer, not shared. It looks like installing nix as single-user isn't very happy about existing single-user installations and breaks somewhere™️. To avoid potential bigger issues I'm no longer sharing that volume with every container using the feature. 

My local tests are still running, but it looks promising so far.

Reply:
Looks like it worked 🎉